packages/@ember/-internals/runtime/lib/mixins/array.js

```diff
@@ -874,18 +874,27 @@ const ArrayMixin = Mixin.create(Enumerable, {
     - `index` is the current index in the iteration.
     - `array` is the array object itself.
 
-    It must return a truthy value (i.e. `true`) to include an item in the
-    results. Any non-truthy return value will discard the item from the
-    results.
+    A return of a falsey value (i.e. `false`) will continue evaluating items in
+    the array until it reaches the end or encounters a truthy (i.e. `true`)
+    return value. If the evaluation reaches the end if the array the result
+    will be `false`. Conversely any evaluation that ends the loop (via
+    returning truthy) will result in `true`.
 
     Note that in addition to a callback, you can also pass an optional target
-    object that will be set as `this` on the context. This is a good way
-    to give your iterator function access to the current object.
+    object that will be set as `this` on the context. It can be a good way
+    to give your iterator function access to the current object in cases where
+    an ES6 arrow function is not appropriate.
 
     Usage Example:
 
     ```javascript
-    if (people.any(isManager)) {
+    let includesManager = people.any(function(person) {
```

Comment:
Maybe include the definition of `people` so that it's clearer what the code sample is achieving.